### DISCUSSION TITLE
Delete confirmation

### DISCUSSION THREAD

**Discussion by OCRAMIS** [Power-Components/livewire-powergrid#242] [Q&A] [answered]
Hi,

The **Delete** button needs confirmation alert modal. Otherwise, rows can get deleted by accident. Is there any standard method to create a **Delete Conformation Modal**?

Thank you in advance.

**Comment by dansysanalyst** [comment]
Hi @OCRAMIS,

I think the best way to proceed is using another Livewire component to perform the confirmation for you.

Your `delete button` calls another component which performs the confirmation and interact with your table using events.

@luanfreitasdev recently added an example calling a modal in the demo repository https://github.com/Power-Components/powergrid-demo. It is not the complete solution, but I hope it helps.

Commit: https://github.com/Power-Components/powergrid-demo/commit/c0702fe64079eb3196e2caf32b49e0f3742ea14e

**Comment by OCRAMIS** [comment]
Hi @dansysanalyst, 

Thank you for your response and marvellous support for the great repo.

I did follow the instructions as per @luanfreitasdev.
It doesn't seem to be working with Jetstream modal ( throwing errors as if it goes outside the Users component ).
However, I have achieved the following up to now in order to complete the project ( I am not satisfied with my project though ).

1)
**Users component:** ( I go to the edit page Instead of deleting it on the users.index page )

        final class Users extends PowerGridComponent
        {
            use ActionButton;
        
            protected function getListeners()
            {
                return array_merge(
                    parent::getListeners(), ['deleteUser']);
            }
        
           /*
            public function deleteUser($data)
            {
                //dd($data);
                $this->dispatchBrowserEvent('delete-user', $data);
            }
            */


    public function actions(): array
    {
       return [
           Button::add('edit')
               ->target('')
               ->caption('<i class="icon ion-md-create"></i>')
               ->class('bg-indigo-500 cursor-pointer text-white px-1 py-0 m-1 rounded text-sm')
               ->route('users.edit', ['user' => 'id']),
           /*
           Button::add('destroy')
               ->target('')
               ->caption('<i class="icon ion-md-trash"></i>')
               ->class('bg-red-700 cursor-pointer text-white px-1 py-0 m-1 rounded text-sm')
               ->emit('deleteUser', ['user' => 'id'])
               //->openModal('confirmingUserDeletion', ['user' => 'id'])
               //->route('users.destroy', ['user' => 'id'])
               ->method('delete'),
            */

**/users/edit.blade.php** ( I use a for with CSRF ) - This works alright although you need to go to the edit page first )

                             <x-form action="{{ route('users.destroy', $user) }}" method="POST" onsubmit="return confirm('{{ __('crud.common.are_you_sure') }}')">
                        @csrf @method('DELETE')
                        <x-jet-secondary-button type="submit" class="text-red-600 bg-gray-50 hover:bg-red-600 hover:text-white float-right">
                            <i class="mr-1 icon ion-md-trash text-white-600"></i>
                            @lang('crud.common.delete')
                        </x-jet-secondary-button>
                    </x-form>

2)
**_In fact what I would like to achieve now is the following. And would appreciate any suggestions._**

a) 
Create an Action Button with `<x-form></x-form>`,  wherein I can use CSRF
b)
If possible to Bulk Select / Bulk Delete on users.index ( listing ) page.

Kind Regards.

**Reply by luanfreitasdev** [reply]
https://github.com/Power-Components/powergrid-demo

Hi @OCRAMIS , can you take a look at this modal again please? I just put a confirmation before deleting, inside the delete function you can instantiate the class with the delete rule too and put the proper permissions

Thanks

**Reply by OCRAMIS** [reply]
Hi @luanfreitasdev, 

Thanks for the update and making it very clear. Delete confirmation is OK now. 

How about the bulk delete. Can we do it something similar to what FilamentAdmin has on top or even a separate delete button next to the csv download?

This great repo can easily be converted to an Admin Panel if the bulk delete feature can be added.

All the Best.

**Reply by luanfreitasdev** [reply]
It is now possible to do some things using

take a look again at today's update on powergrid-demo

![image](https://user-images.githubusercontent.com/33601626/151716059-fc13a36a-4fbb-4e3c-8053-08d3760ad6e8.png)

**Comment by OCRAMIS** [comment, marked as answer]
Hi @luanfreitasdev,

Thank you very much for your excellent ( I have no other better words to explain ) support for this great repo.
I have tested the demo and it works perfect.
Now with this feature PowerGrid repo can be developed as a livewire admin panel.
I am working on it for my own projects.
Will let you know if I come across any other issues/requests.

Stay Blessed!

Best Regards.

**Comment by ashek1412** [comment]
Hi @luanfreitasdev ,
I cant implement single row delete action. modal is not opening.
![Capture](https://user-images.githubusercontent.com/18645594/183487294-f8abf8cb-c7f9-4ddf-9005-976f4a9d5823.PNG)

**Reply by luanfreitasdev** [reply]
I didn't understand the problem
